### ISSUE TITLE
 Memory leak after upgrading from 0.70.4 to 0.72.1

### ISSUE THREAD

**Issue by julienvey** [RocketChat/Rocket.Chat#12880]
### Description:

We recently upgraded our rocketchat instance from version 0.70.4 to 0.72.1. Since then, we are experiencing memory leaks on daytime (when people are using it). 

I'm not sure what kind of info I can give you to help understand the issue, so please let me know

### Actual behavior:

During daytime, when people are using it, memory is leaking from the rocketchat docker instance (memory graph below). When people are out of office, memory is fine

### Server Setup Information:

- Version of Rocket.Chat Server: 0.72.1
- Operating System: Debian Jessie
- Deployment Method: docker on kubernetes (rocketchat and mongo in separated pods)
- Number of Running Instances: 1
- DB Replicaset Oplog: Disabled
- NodeJS Version: v8.11.3
- MongoDB Version: v3.6.5

### Additional context

![capture d ecran 2018-12-07 a 11 18 17](https://user-images.githubusercontent.com/455944/49641909-dd5c4580-fa11-11e8-8926-581503a43863.png)

- Red line is the upgrade
- before the upgrade, memory consumption was flat
- Memory is dropping when the pod is being killed, and restarted

**Comment by himpierre** [comment]
No memory leak on my systems. Running rocketchat 0.72.1 on 4 machines. Gentoo and Ubuntu. But i do not use docker or kubernetes. So i suppose it's related to that.

**Comment by sjoerdmaessen** [comment]
This might be related to an issue I'm having although I'm not sure: https://forums.rocket.chat/t/sudden-100-cpu-usage-causing-non-responsive-chat-till-service-restart/2631 but I have seen a lot of issues with the Node version we are both currently running regards memory. v8.12 should be a lot better.

**Comment by julienvey** [comment]
We tried to update to the following configuration, but the problem still remains

### Server Setup Information:
Operating System: Debian Stretch
NodeJS Version: v8.14.0

**Comment by tlhonmey** [comment]
This also seems to be a problem on CentOS with node version v8.14.0.  Leak rate seems to correlate with message volume, but that could be a coincidence.

**Comment by julienvey** [comment]
To add information about the issue. We recently switched to a multiple instances rocketchat installation and we no longer see any memory leak. So the problem seems to happen only when using a single rocketchat instance (or a single mongo database ?)

![Capture d’écran 2019-04-23 à 13 28 22](https://user-images.githubusercontent.com/455944/56577680-dc484100-65cb-11e9-8c81-02ff160e1d71.png)

**Comment by ankar84** [comment]
> So the problem seems to happen only when using a single rocketchat instance

No. Multi Instance deployment affected ny memory leak too https://github.com/RocketChat/Rocket.Chat/issues/17473#issuecomment-636642842
